lib/services/backup_service.dart

```diff
@@ -35,17 +38,37 @@ class BackupService extends ChangeNotifier {
     };
   }
 
-  Future<AppBackup?> readBackupFromFile() {
-    return _fileService //
-        .selectFile()
-        .then((file) => file?.readAsString() ?? Future(() => null))
-        .then((value) => (value != null) ? AppBackup.fromString(value) : null);
+  Future<AppBackup?> readBackupFromFile() async {
+    late String? backupDestination;
+    try {
+      final (file, filePickerId) = await _fileService.selectFile();
+      backupDestination = filePickerId;
+      if (file != null) {
+        final fileContent = await file.readAsString();
+        _analyticManager.trackRecoverFromFile(success: true, source: PageSource.get_started, backup: backupDestination);
```

Comment:
This tracking is premature and incorrect. At the very least, needs to be moved after creation of `AppBackup`. But even then, it doesn't really mean that recover has been successful - it just means we were able to read the backup. Recovering is yet another matter, and it cannot be tracked in this service alone.

Reply:
The recovering logic by far was:
1. Check if there's any backup content in File System
2. Load the content and create AppBackup
3. Pairing with WalletBackup

Currently, `trackRecoverFromFile` is tracking stage 2. And `trackPairingDevice` is tracking stage 3 with `type: recovered`. IMHO, its still correct.